include/seqan3/alignment/matrix/detail/alignment_trace_matrix_full.hpp

```diff
@@ -123,12 +122,32 @@ class alignment_trace_matrix_full :
 
         if constexpr (!coordinate_only)
         {
-            matrix_base_t::data.resize(matrix_base_t::num_rows * matrix_base_t::num_cols);
+            // Allocate the matrix here.
+            matrix_base_t::data = typename matrix_base_t::pool_type{number_rows{matrix_base_t::num_rows},
+                                                                    number_cols{matrix_base_t::num_cols}};
             matrix_base_t::cache_left.resize(matrix_base_t::num_rows, initial_value);
         }
     }
     //!\}
 
+    /*!\brief Returns a trace path starting from the given coordinate and ending in the cell with
+     *        seqan3::detail::trace_directions::none
+     * \param[in] trace_begin A seqan3::matrix_coordinate pointing to the begin of the trace to follow.
+     * \returns A std::ranges::subrange over the corresponding trace path.
+     */
+    auto trace_path(matrix_coordinate const & trace_begin)
+    {
+        using matrix_iter_t = std::ranges::iterator_t<typename matrix_base_t::pool_type>;
```

Comment:
you do not assert here that the configuration was not `coordinate only` like you did below in the banded case.